test/unit/BPool/BPool_SwapExactAmountIn.tree

```diff
@@ -0,0 +1,28 @@
+BPool::SwapExactAmountIn
+├── when reentrancy lock is set
+│   └── it should revert
+├── when pool is not finalized
+│   └── it should revert
+├── when token in is not bound
+│   └── it should revert
+├── when token out is not bound
+│   └── it should revert
+├── when token amount in exceeds max allowed ratio
+│   └── it should revert
+├── when spot price before swap exceeds maxPrice
+│   └── it should revert
+├── when calculated token amount out is less than minAmountOut
+│   └── it should revert
+├── when spot price after swap exceeds maxPrice
+│   └── it should revert
+├── when spot price before swap exceeds token ratio after swap
+│   └── it should revert
```

Comment:
missing `BPool_SpotPriceAfterBelowSpotPriceBefore` revert (also unreachable but perhaps mockable)

Reply:
mathematically it seems unreachable, and in order to mock it I should:

- make `calcSpotPrice` virtual
- manually override it in the mocking for `BPool` to force an external call
- insert a mock for the call with the *second* set of parameters to return a value lower than for the first one. Thankfully the parameters are different in each invocation, so a regular `mockCall` can take care of it and we wouldn't have to make a custom override counting calls.

Adding a skipped test to be explicit about it